numpy/core/code_generators/generate_umath.py

```diff
@@ -676,7 +676,7 @@ def english_upper(s):
           docstrings.get('numpy.core.umath.cos'),
           None,
           TD('e', f='cos', astype={'e':'f'}),
-          TD('f', simd=[('fma', 'f'), ('avx512f', 'f')]),
+          TD('f', dispatch=[('loops_trigonometric', 'fd')]),
```

Comment:
Shouldn't this be
```suggestion
          TD('fd', dispatch=[('loops_trigonometric', 'fd')]),
```

(and below as well)

Reply:
Ahh, 32-bit only. Thanks.

Reply:
I fixed it. just a typo not affect on the code path since TD type doesn't has `d`, anyway dispatch should only contains `f` and double-precision defined by the row after.